Sources/AsyncHTTPClient/AsyncAwait/HTTPClientResponse.swift

```diff
@@ -138,6 +138,19 @@ extension HTTPClientResponse {
     }
 }
 
+@available(macOS 10.15, iOS 13.0, watchOS 6.0, tvOS 13.0, *)
+extension HTTPClientResponse {
+    /// Response body as `ByteBuffer`.
+    public var bytes: ByteBuffer {
+        get async throws {
+            let expectedBytes = headers
+                .first(name: "content-length")
+                .flatMap(Int.init) ?? 1024 * 1024
+            return try await body.collect(upTo: expectedBytes)
```

Comment:
I like this helper, but I think it might be a bit better as a function. In particular I'm very reluctant to let users avoid providing the default maximum body size: just hard-coding this to 1MB is probably going to cause some confusion.

Reply:
Added 2 new tests

Reply:
I'm not sure we should automatically depend on "content-length" here. A malicious client may send just peta bytes of data, which can bring down the client.

Reply:
If you're getting an image from some URL you probably don't know is its size 1mb or 20mb. If you'll pass too small `upTo` value there will be a crash.

Here's an example that gives `Swift/ErrorType.swift:200: Fatal error: Error raised at top level: NIOCore.NIOTooManyBytesError()` with 1.21.0:
```swift
let request = HTTPClientRequest(url: "https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png")
let response = try await HTTPClient.shared.execute(request, timeout: .seconds(30))
let buffer = try await response.body.collect(upTo: 10)
```

If you don't know the data size it makes sense to provide some big value (like 1012 * 1012 * 100 for 100MB for example), or to use "content-length" header that tells you the size of the data.

If you don't trust the source it's still possible to provide `upTo` param in the `public func bytes(upTo maxBytes: Int? = nil) async throws -> ByteBuffer` method above.

Do you have a better idea? 🙂

Reply:
A better version might be to provide a maximum value the user is willing to accept here, and then use content-length _or_ the maximum value.

Reply:
Thanks Lukasa. I've updated the PR, made `upTo` param required for both helper functions as you suggested. If the content-length is larger than the max there will be an error thrown. Also updated the tests.